source/initial_composition/crust1.cc

```diff
@@ -0,0 +1,225 @@
+/*
+  Copyright (C) 2017 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/initial_composition/crust1.h>
+#include <aspect/geometry_model/interface.h>
+#include <aspect/material_model/crust1.h>
+#include <aspect/geometry_model/spherical_shell.h>
+#include <aspect/utilities.h>
+#include <aspect/simulator_access.h>
+#include <fstream>
+#include <iostream>
+#include <array>
+
+namespace aspect
+{
+  namespace InitialComposition
+  {
+
+    template <int dim>
+    double
+    Crust1<dim>::
+    initial_composition (const Point<dim> &position,
+                         const unsigned int compositional_index) const
+    {
+      AssertThrow(this->introspection().compositional_name_exists("crust1_density"),
+                  ExcMessage("The initial composition plugin `crust1' did not find a "
+                             "compositional field called `crust1_density' to initialize. Please add a "
+                             "compositional field with this name."));
+
+      AssertThrow(dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) != nullptr,
+                  ExcMessage("The initial composition plugin `crust1' only works with "
+                             "the 'SphericalShell' geometry model."));      
+
+      AssertThrow(dynamic_cast<const MaterialModel::Crust1<dim>*> (&this->get_material_model()) != nullptr,
+                  ExcMessage("The initial composition plugin `crust1' only works with "
+                             "the 'Crust1' material model."));
+
+      const unsigned int crust1_index = this->introspection().compositional_index_for_name("crust1_density");
+      if (compositional_index == crust1_index)
+        {
+          
+          // Get coordinates (surface elevation, longitude (0-360), latitude (-90 - 90))
+          std::array<double,dim> wcoord = Utilities::Coordinates::WGS84_coordinates(position);
```

Comment:
If a spherical shell is being used, wouldn't cartesian_to_spherical be a better choice than WGS84?
Also, const?

Reply:
Good catch. The variable name should probably be changed to `scoord` for consistency as well, e.g.:
```suggestion
          const std::array<double,dim> scoord = Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
```